app/models/miq_queue_worker_base/runner.rb

```diff
@@ -62,9 +66,23 @@ def get_message_via_sql
     end
   end
 
+  def get_message_via_miq_messaging
+    @message_queue ||= Queue.new
+    @listener_thread = nil if @listener_thread && !@listener_thread.alive?
+    @listener_thread ||= Thread.new do
+      MiqQueue.messaging_client(self.class.name)
+              .subscribe_topic(:service => self.class.miq_messaging_service, :persist_ref => @worker.guid) do |msg|
+        @message_queue << msg
+      end
+    end
+    @message_queue.pop unless @message_queue.empty?
+  end
+
   def get_message
     if dequeue_method_via_drb? && @worker_monitor_drb
       get_message_via_drb
+    elsif dequeue_method_via_miq_messaging?
```

Comment:
do we need to do similar type things as drb if we have many workers listening to kafka queue?